### PR TITLE
lazy deserialization of event to map,multimap,queue,topic,list,set

### LINKED ISSUE #6582: Lazy Deserialization required on events processing
If 2 clients are listening on entryEvicted of EntryEvictedListener only, a map.put on one client results in a callback ClientMapProxy$ClientMapEventHandler.creteEntryEvent() on other clients, which further leads to deserialisation of the entry on those clients. This causes GC and excess CPU issues on the clients which are not even actively making map calls. 

It requires lazy deserialization on clients when the event arrives. This will prevent excess CPU and memory usage spikes.

### PR DESCRIPTION
**Pull request by sancar**
fixes #6582

### CODE REVIEW

File: hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMultiMapProxy.java [hunk 1]
```diff
@@ -478,16 +479,9 @@ private MapEvent createMapEvent(int eventType, int numberOfAffectedEntries, Memb
...
-            V oldValue = null;
-            V mergingValue = null;
-            if (includeValue) {
-                value = toObject(valueData);
-                oldValue = toObject(oldValueData);
-                mergingValue = toObject(mergingValueData);
-            }
-            K key = toObject(keyData);
-            return new EntryEvent<K, V>(name, member, eventType, key, oldValue, value, mergingValue);
+            return new DataAwareEntryEvent(member,
```
Comment (pveentjer) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMultiMapProxy.java]: Please rename this class to LazyDeserializingEntryEvent.

**Comment by pveentjer** [comment]
Some tests need to be added to test this behavior.

File: hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientTopicProxy.java [hunk 2]
```diff
@@ -112,12 +113,11 @@ private TopicItemHandler(MessageListener<E> listener) {
...
         public void handle(Data item, long publishTime, String uuid) {
-            final SerializationService serializationService = getContext().getSerializationService();
-            final ClientClusterService clusterService = getContext().getClusterService();
+            SerializationService serializationService = getContext().getSerializationService();
+            ClientClusterService clusterService = getContext().getClusterService();
 
-            E messageObject = serializationService.toObject(item);
             Member member = clusterService.getMember(uuid);
-            Message<E> message = new Message<E>(name, messageObject, publishTime, member);
+            Message<E> message = new DataAwareMessage(name, item, publishTime, member, serializationService);
```
Comment (pveentjer) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientTopicProxy.java]: I would rename this to LazyDeserializingMessage.

**Comment by sancar** [comment]
@pveentjer Do you think we should add a test to check if event has data and not serialized yet, then we called the getValue then it is deserialized? If you dont think it is necessary, other listener tests are already available.

**Comment by sancar** [comment]
Merging this since it is reviewed and merged here. https://github.com/hazelcast/hazelcast/pull/6590
I take my note about test, I will think about how can I implement and send another pr later.

**Pull request merged**